app/native/features/auth/useAuthStore.ts

```diff
@@ -0,0 +1,192 @@
+import { useQueryClient } from "react-query";
+import { service } from "service";
+import isGrpcError from "service/utils/isGrpcError";
+
+export default function useAuthStore() {
+  const [authenticated, setAuthenticated] = usePersistedState(
+    "auth.authenticated",
+    false,
+  );
+  const [jailed, setJailed] = usePersistedState("auth.jailed", false);
+  const [checkedAuthStatus, setCheckedAuthStatus] = usePersistedState(
+    "auth.checkedAuthStatus",
+    false,
+  );
+  const [userId, setUserId] = usePersistedState<number | null>(
+    "auth.userId",
+    null,
+  );
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [flowState, setFlowState] =
+    usePersistedState<SignupFlowRes.AsObject | null>("auth.flowState", null);
+
+  //this is used to set the current user in the user cache
+  //may as well not waste the api call since it is needed for userId
+  const queryClient = useQueryClient();
+
+  const { t } = useTranslation(GLOBAL);
+  const fatalErrorMessage = useRef(t("error.fatal_message"));
+  const authActions = useMemo(
+    () => ({
+      authError(message: string) {
+        setError(message);
+      },
+      clearError() {
+        setError(null);
+      },
+      async logout() {
+        setError(null);
+        setLoading(true);
+        try {
+          await service.user.logout();
+          setAuthenticated(false);
+          setUserId(null);
+          Sentry.setUser({ id: undefined });
+        } catch (e) {
+          Sentry.captureException(e, {
+            tags: {
+              component: "auth/useAuthStore",
+              action: "logout",
+            },
+          });
+          setError(isGrpcError(e) ? e.message : fatalErrorMessage.current);
+        }
+        clearStorage();
+        setLoading(false);
+      },
+      async passwordLogin({
+        username,
+        password,
+        rememberDevice,
+      }: {
+        username: string;
+        password: string;
+        rememberDevice: boolean;
+      }) {
+        setError(null);
+        setLoading(true);
+        try {
+          const auth = await service.user.passwordLogin(
+            username,
+            password,
+            rememberDevice,
+          );
+          setUserId(auth.userId);
+          Sentry.setUser({ id: auth.userId.toString() });
+
+          //this must come after setting the userId, because calling setQueryData
+          //will also cause that query to be background fetched, and it needs
+          //userId to be set.
+          setJailed(auth.jailed);
+          setAuthenticated(true);
+        } catch (e) {
+          Sentry.captureException(e, {
+            tags: {
+              component: "auth/useAuthStore",
+              action: "passwordLogin",
+            },
+          });
+          setError(isGrpcError(e) ? e.message : fatalErrorMessage.current);
+        }
+        setLoading(false);
+      },
```

Comment:
This is quite hard to read and tell what's going on. Maybe you could break out the functions into separate variables and then just call them within the main function to abstract things out a bit?